src/box/field_def.c

```diff
@@ -61,7 +61,7 @@ const uint32_t field_mp_type[] = {
 	/* [FIELD_TYPE_STRING]   =  */ 1U << MP_STR,
 	/* [FIELD_TYPE_NUMBER]   =  */ (1U << MP_UINT) | (1U << MP_INT) |
 		(1U << MP_FLOAT) | (1U << MP_DOUBLE),
-	/* [FIELD_TYPE_DOUBLE]   =  */ 1U << MP_DOUBLE,
+	/* [FIELD_TYPE_DOUBLE]   =  */ (1U << MP_DOUBLE) | (1U << MP_UINT) | (1U << MP_INT),
```

Comment:
I do understand that that was my patch suggested, but, in any case, I've discovered that during later discussions.

I believe, that in this line we should also allow `MP_FLOAT` data types. 

P.S.
I don't know yet how to generate `MP_FLOAT` messagepack data from generic Tarantool tuple.